sql/src/main/java/io/crate/operation/projectors/DistributingDownstreamFactory.java

```diff
@@ -112,4 +110,22 @@ public BatchConsumer create(NodeOperation nodeOperation,
             pageSize
         );
     }
+
+    /**
+     * @return bucketIdx (= idx of localNode in nodeIds)
+     *
+     * <p>
+     * This needs to be deterministic across all nodes involved in a query execution.
+     *
+     * Downstreams assume that each upstream uses a *different* bucketIdx,
+     * but they don't care which node ends up with which bucketIdx.
+     * </p>
+     *
+     * See {@link io.crate.jobs.PageBucketReceiver}
+     */
+    private int getBucketIdx(Collection<String> nodeIds) {
+        ArrayList<String> server = new ArrayList<>(nodeIds);
+        server.sort(null);
+        return Math.max(server.indexOf(clusterService.localNode().getId()), 0);
```

Comment:
Would it make sense to use a `SortedSet` for the node ids? And perhaps create a method `getBucketIdx(ClusterService cs)` on the `NodeOperation` which returns the idx?

Reply:
I'd keep it as it is - I doubt that there is a big difference between SortedSet/ArrayList in this case as the number of nodeIds is usually rather small - and with SortedSet there would be increased memory usage.

I'd also keep it here for now as it's the only place where it's used. I just wanted to improve the docs/get rid of the TODO, not change the behaviour in any other way =)

Reply:
Ok, just a suggestion. It is definitely an improvement as it is.